test/Twig/Tests/EnvironmentTest.php

```diff
@@ -145,10 +145,17 @@ public function testExtensionsAreNotInitializedWhenRenderingACompiledTemplate()
         // force compilation
         $twig = new Twig_Environment($loader = new Twig_Loader_Array(array('index' => '{{ foo }}')), $options);
         $cache = $twig->getCacheFilename('index');
-        if (!is_dir(dirname($cache))) {
-            mkdir(dirname($cache), 0777, true);
+        $dir = dirname($cache);
+        $created = false;
+        if (!is_dir($dir)) {
+            if (false === $created = @mkdir($dir, 0777, true)) {
+                $this->markTestIncomplete(sprintf('Failed to create test cache directory at "%s".', $dir));
+            }
+        }
+
+        if (false === @file_put_contents($cache, $twig->compileSource('{{ foo }}', 'index'))) {
```

Comment:
please don't put the ``$twig->compileSource('{{ foo }}', 'index')`` expression inside the silenced statement, as we would loose any error triggered in the Twig code here